### PR TITLE
refactor region and resource group

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/.idea/gradle.xml [hunk 1]
```diff
@@ -12,6 +12,9 @@
           <set>
             <option value="$PROJECT_DIR$" />
             <option value="$PROJECT_DIR$/azure-intellij-plugin-lib" />
+            <option value="$PROJECT_DIR$/azure-intellij-resource-connector-aad" />
+            <option value="$PROJECT_DIR$/azure-intellij-resource-connector-lib" />
+            <option value="$PROJECT_DIR$/azure-intellij-resource-connector-mysql" />
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/.idea/gradle.xml:17]: do we need to update and push the file under ".idea" into git repo ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/.idea/misc.xml [hunk 2]
```diff
@@ -38,9 +41,16 @@
           <item index="8" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.compatqual.NonNullType" />
           <item index="9" class="java.lang.String" itemvalue="com.android.annotations.NonNull" />
           <item index="10" class="java.lang.String" itemvalue="com.microsoft.azuretools.azurecommons.helpers.NotNull" />
+          <item index="11" class="java.lang.String" itemvalue="io.reactivex.annotations.NonNull" />
+          <item index="12" class="java.lang.String" itemvalue="lombok.NonNull" />
+          <item index="13" class="java.lang.String" itemvalue="io.reactivex.rxjava3.annotations.NonNull" />
+          <item index="14" class="java.lang.String" itemvalue="org.eclipse.jdt.annotation.NonNull" />
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/.idea/misc.xml:47]: same above

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/redis/CreateRedisCacheForm.java [hunk 3]
```diff
@@ -77,26 +80,25 @@
...
     private JTextField txtRedisName;
-    private JComboBox<SubscriptionDetail> cbSubs;
+    private com.microsoft.azure.toolkit.intellij.appservice.subscription.SubscriptionComboBox cbSubs;
     private JRadioButton rdoCreateNewGrp;
     private JTextField txtNewResGrp;
     private JRadioButton rdoUseExist;
     private JComboBox<String> cbUseExist;
     // TODO(qianjin) : use AzureComboBox
-    private JComboBox<Region> cbLocations;
+    private com.microsoft.azure.toolkit.intellij.appservice.region.RegionComboBox cbLocations;
```
[the remark this answers is not in the document]
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/redis/CreateRedisCacheForm.java:89]: is RegionComboBox belong to appsevice lib? --> why not common-lib ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/redis/CreateRedisCacheForm.java [hunk 4]
```diff
@@ -77,26 +80,25 @@
     // Widgets
     private JPanel pnlContent;
     private JTextField txtRedisName;
-    private JComboBox<SubscriptionDetail> cbSubs;
+    private com.microsoft.azure.toolkit.intellij.appservice.subscription.SubscriptionComboBox cbSubs;
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/redis/CreateRedisCacheForm.java:83]: is SubscriptionComboBox belong to appsevice lib? --> why not common-lib ?

**Comment by andxu** [comment]
replaced by normal PR: https://github.com/microsoft/azure-tools-for-java/pull/5251

**Pull request closed without merging**